src/openms/source/QC/FragmentMassError.cpp

```diff
@@ -192,9 +192,15 @@ namespace OpenMS
     // WRITE PPM ERROR IN PEPTIDEHIT
     //-----------------------------------------------------------------------
     pep_id.getHits()[0].setMetaValue(Constants::UserParam::FRAGMENT_ERROR_PPM_METAVALUE_USERPARAM, ppms);
-    pep_id.getHits()[0].setMetaValue(Constants::UserParam::FRAGMENT_ERROR_PPM_METAVALUE_USERPARAM + "_variance", Math::variance(ppms.begin(), ppms.end(), Math::mean(ppms.begin(), ppms.end())));
     pep_id.getHits()[0].setMetaValue(Constants::UserParam::FRAGMENT_ERROR_DA_METAVALUE_USERPARAM, dalton);
-    pep_id.getHits()[0].setMetaValue(Constants::UserParam::FRAGMENT_ERROR_DA_METAVALUE_USERPARAM + "_variance", Math::variance(dalton.begin(), dalton.end(), Math::mean(dalton.begin(), dalton.end())));
+    if (!ppms.empty())
```

Comment:
can you check if variance works if only one value is present?

https://github.com/OpenMS/OpenMS/blob/55598cf39c73a2092365fbdb51e63defa9b29c6f/src/openms/include/OpenMS/MATH/STATISTICS/StatisticFunctions.h#L292

Reply:
Well, it doesn't throw an error. It just returns `-nan(ind)`. But maybe I should still prevent that from happening...